boltons/dictutils.py

```diff
@@ -693,4 +693,104 @@ def __reversed__(self):
             yield curr[KEY]
             curr = curr[PREV]
 
+
+class SentinelInt(int):
+    pass
+
+
+class FrozenDictBase(tuple):
+
+    def __getitem__(self, key):
+        if isinstance(key, SentinelInt):
+            return tuple.__getitem__(self, key)
+        else:
+            try:
+                return getattr(self, str(key.__hash__()))()
+            except AttributeError:
+                raise KeyError(key)
+            except IndexError:
+                raise KeyError(key)
+
+    def __iter__(self):
+        idx = 0
+        while True:
+            try:
+                yield self[SentinelInt(idx)][0]
+            except IndexError:
+                break
+            idx += 1
+
+    def items(self):
+        """Returns a list of tuples of data"""
+        return [(x, self[x]) for x in self]
+
+    def iteritems(self):
+        """Iterates, returning key, value pairs as a tuple"""
+        idx = 0
+        while True:
+            try:
+                yield self[SentinelInt(idx)][0], self[SentinelInt(idx)][1]
+            except IndexError:
+                break
+            idx += 1
```

Comment:
This isn't a subclass of a dict so the super calls would be going to the tuple class which won't return the same things.